### PR TITLE
Add io.github.yuki_iptv.yuki-iptv

### PR DESCRIPTION
**Pull request by faveoled**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [https://github.com/yuki-iptv/yuki-iptv/issues/38](https://github.com/yuki-iptv/yuki-iptv/issues/38)
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by faveoled** [comment]
bot, build io.github.yuki_iptv.yuki-iptv

**Comment by faveoled** [comment]
bot, build io.github.yuki_iptv.yuki-iptv

File: io.github.yuki_iptv.yuki-iptv.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      - cp -r usr/* /app/
+      - rm -rf usr
+      
+      - install -Dm644 /app/share/metainfo/yuki-iptv.appdata.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|type="desktop"|type="desktop-application"|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<id>yuki-iptv.desktop</id>|<id>io.github.yuki_iptv.yuki-iptv</id>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<launchable type="desktop-id">yuki-iptv.desktop</launchable>|<launchable type="desktop-id">io.github.yuki_iptv.yuki-iptv.desktop</launchable>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      
+      - install -Dm644 /app/share/icons/hicolor/scalable/apps/yuki-iptv.svg /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.svg
+      - install -Dm644 /app/share/applications/yuki-iptv.desktop /app/share/applications/${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline io.github.yuki_iptv.yuki-iptv.yml:82]: just use `rename-desktop-file` and then you can remove lines 77, 78 as well.

File: io.github.yuki_iptv.yuki-iptv.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+      - cp -r usr/* /app/
+      - rm -rf usr
+      
+      - install -Dm644 /app/share/metainfo/yuki-iptv.appdata.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|type="desktop"|type="desktop-application"|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<id>yuki-iptv.desktop</id>|<id>io.github.yuki_iptv.yuki-iptv</id>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<launchable type="desktop-id">yuki-iptv.desktop</launchable>|<launchable type="desktop-id">io.github.yuki_iptv.yuki-iptv.desktop</launchable>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      
+      - install -Dm644 /app/share/icons/hicolor/scalable/apps/yuki-iptv.svg /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.svg
```
Comment (hfiguiere) [inline io.github.yuki_iptv.yuki-iptv.yml:81]: use `rename-icon` and then you can remove line 85.

File: io.github.yuki_iptv.yuki-iptv.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      - tar -xvf data.tar.xz
+     
+      # fix for later dir copy error: cannot overwrite non-directory with directory
+      - mv usr/share/locale/pt_BR/LC_MESSAGES/yuki-iptv.mo /app/share/locale/pt_BR/LC_MESSAGES/
+      - rm -rf usr/share/locale/pt_BR
+
+      - cp -r usr/* /app/
+      - rm -rf usr
+      
+      - install -Dm644 /app/share/metainfo/yuki-iptv.appdata.xml /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
```
Comment (hfiguiere) [inline io.github.yuki_iptv.yuki-iptv.yml:76]: use `rename-appdata-file`.

File: io.github.yuki_iptv.yuki-iptv.yml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      - sed -i 's|type="desktop"|type="desktop-application"|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<id>yuki-iptv.desktop</id>|<id>io.github.yuki_iptv.yuki-iptv</id>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      - sed -i 's|<launchable type="desktop-id">yuki-iptv.desktop</launchable>|<launchable type="desktop-id">io.github.yuki_iptv.yuki-iptv.desktop</launchable>|g' /app/share/metainfo/${FLATPAK_ID}.metainfo.xml
+      
+      - install -Dm644 /app/share/icons/hicolor/scalable/apps/yuki-iptv.svg /app/share/icons/hicolor/512x512/apps/${FLATPAK_ID}.svg
+      - install -Dm644 /app/share/applications/yuki-iptv.desktop /app/share/applications/${FLATPAK_ID}.desktop
+
+      - desktop-file-edit --set-key Exec --set-value "/app/bin/run.sh" ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline io.github.yuki_iptv.yuki-iptv.yml:84]: the executable value sould NOT be a full path.

**Comment by faveoled** [comment]
bot, build io.github.yuki_iptv.yuki-iptv

**Pull request closed without merging**